src/Shared/LayoutRenderers/AspNetSessionValueLayoutRenderer.cs

```diff
@@ -93,36 +89,33 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
             }
 
             var context = HttpContextAccessor.HttpContext;
-            var contextSession = context.TryGetSession();
-            if (contextSession == null)
-                return;
-
-#if !ASP_NET_CORE
-            var value = PropertyReader.GetValue(item, contextSession, (session,key) => session.Count > 0 ? session[key] : null, EvaluateAsNestedProperties);
-#else
-            //because session.get / session.getstring also creating log messages in some cases, this could lead to stackoverflow issues. 
-            //We remember on the context.Items that we are looking up a session value so we prevent stackoverflows
-            if (context.Items == null || (context.Items.Count > 0 && context.Items.ContainsKey(NLogRetrievingSessionValue)))
+            if (context == null)
             {
                 return;
             }
 
-            context.Items[NLogRetrievingSessionValue] = bool.TrueString;
-
-            object value;
-            try
-            {
-                value = PropertyReader.GetValue(item, contextSession, (session, key) => GetSessionValue(session, key), EvaluateAsNestedProperties);
-            }
-            finally
+            var contextSession = context?.TryGetSession();
+            if (contextSession == null)
             {
-                context.Items.Remove(NLogRetrievingSessionValue);
+                return;
             }
-#endif
-            if (value != null)
+
+            using (var reEntry = new RendererReEntrantManager(context))
             {
-                var formatProvider = GetFormatProvider(logEvent, Culture);
-                builder.AppendFormattedValue(value, Format, formatProvider, ValueFormatter);
+                if (!reEntry.IsLockAcquired)
+                {
+                    InternalLogger.Error(
```

Comment:
Can this become `InternalLogger.Debug` and without using string-interpolation ? (Skip logging `logEvent.Message`)

Reply:
Complete